---
title: "1. Preprocessing"
format: html
editor: visual
---

## Introduction

Generally, the preprocessing workflow for proteomics data encompasses several key steps:

-   **Raw data acquisition:** data collection through mass spectometry techniques.

-   **Database search**: utilisation of databases to match acquired spectra with known protein sequences.

-   **Quantification:** measurement of protein abundance levels.

-   **Normalisation:** normalise data to correct for technical variability (e.g. total ion current, median normalisation).

-   **Batch effect removal:** mitigation of system variation introduced by different experimental batches.

-   **Log transformation:** application of logarithmic transformation to stabilise variance.

-   **Missing value handling:** addressing and imputing missing values in the dataset (e.g. impute with low values for missing proteins or remove proteins with high missingness).

## Data

The data used in this example has been taken from ([Carlyle et al., 2021](https://www.sciencedirect.com/science/article/pii/S0197458021001275?via%3Dihub#sec0002)).

This paper used multiplexed liquid chromatography mass-spectrometry to quantify enriched synaptic proteins from the parietal association cortex of 100 subjects with contrasting levels of AD pathology and cognitive performance.

The protein intensities have been normalised via a multi-step process. Intensities were first normalised using the intensity measured for the bridge sample. An additional normalisation step performed by normalising the protein intensities measured for each sample by the global median of the per-sample median protein intensities.

### Load libraries

```{r, results="hide", message=FALSE}
library(tidyverse)
library(openxlsx)
library(broom)
library(gridExtra)
```

### Load TMT data

Import the file containing the median-normalised protein quantification data. The observations are signal intensity measurements from the mass spectrometer, and these intensities relate to the amount of protein.

```{r}
TMT_data = read.xlsx("../data/Table_S3_Carlyle_TMT_synaptosomes_quants.xlsx", rowNames = FALSE)

knitr::kable(head(TMT_data))
```

```{r, message=FALSE}
sample_key = read_csv("../data/Table_S1_individual_sample_demographics.csv")

knitr::kable(head(sample_key))
```

### Exploratory data analysis

#### Missingness?

Inspect the missingness of the TMT dataset by counting the number of `NA`s in every non-pooled sample.

```{r}
# Select relevant data (non-pooled samples)
to_count = select(TMT_data, UniprotAccession, A1:J9)
# Assign protein names from UniprotAccession to colnames variable
colnames = to_count$UniprotAccession
# Transpose the dataset - makes proteins the columns and samples as rows
to_count = t(to_count[ ,-c(1)])
# Assign protein names as column headers
colnames(to_count) = colnames
# Convert to dataframe
to_count = as.data.frame(to_count)

knitr::kable(head(to_count))
```

Count NAs for each protein.

```{r}
# Count NAs in each column
# map() applies a function to each column of the dataframe
N_Nas = map(to_count, ~sum(is.na(.)))
# Convert the list output from map() into a simple vector of NA counts
NAs = unlist(N_Nas)

knitr::kable(head(NAs))
```

Transform the results for plotting.

```{r}
# Convert to dataframe
NAs = as.data.frame(as.matrix(NAs))
# Move row names into a column and rename columns
NAs = rownames_to_column(NAs)
colnames(NAs) = c("UniprotAccession" ,"NA_count")

knitr::kable(head(NAs))
```

Create plot theme.

```{r}
my_theme = function() {
  theme_minimal() +
    theme(axis.line = element_line(colour = "black"),  # Add axis lines
          axis.ticks = element_line(colour = "black"),  # Add axis ticks
          plot.title = element_text(face = "bold", hjust=0.5), # Adjust plot title
          axis.title.y = element_text(margin = margin(r = 15), # Adjust y-axis title position
                                      size = 13), # Adjust y-axis title size
          axis.text.x = element_text(size = 10), # Increase x-axis text size
          axis.text.y = element_text(size = 10) # Increase y-axis text size
    ) 
}
```

Plot a histogram showing the distribution of missing data across proteins. For example, a large peak at `NA_count = 0` indicates many proteins have no missing data.

```{r}
ggplot(data = NAs, aes(x=NA_count)) +
  geom_histogram(binwidth = 1) +
  scale_x_continuous(breaks=seq(0,100, by=10)) +
  my_theme()
```

Calculate how many proteins have complete data (no missing values) in this dataset.

```{r}
# Calculate total number of proteins (rows) in the dataset
n_total = nrow(NAs)
n_total # 9560 proteins
```

```{r}
# Calculate the number of proteins (rows) with no missing values
n_complete = sum(NAs$NA_count == 0)
n_complete # 4954 complete proteins
```

Out of the 9560 proteins that were detected in at least one sample, 4954 proteins were quantified in every (non-pooled) sample.

#### Log-transformation?

Check if the data will need to be log-transformed.

Data should be log-transformed if it:

-   Is right-skewed (not normally distributed).

    -   Proteomics data is often right-skewed because a small number of proteins have very high abundance and most proteins are low in abundance.

    -   Many statistical methods (e.g. linear regression and t-tests) assume normal distribution.

-   Contains large ranges.

    -   Proteomics data often spans several orders of magnitude, where some proteins have extremely high abundances, while many others are low in abundance.

    -   Log-transformation compresses the dynamic range of the data.

-   Shows mean-variance dependence.

    -   Without log-transformation, high abundance proteins can dominate the analysis because their variance is larger.

    -   Log-transformation compresses the dynamic range of the data, stabilising variance.

-   Contains extreme values.

    -   Extreme values (outliers) can skew statistical analyses.

    -   Log-transformation reduces their impact by shrinking large values.

Create a dataframe of raw and log2-transformed data.

```{r}
# Select relevant data (non-pooled samples)
to_plot = select(TMT_data, UniprotAccession, A1:J9) %>%
  # Reshape the data from wide format into long format (each row represents a single observation of a protein in sample)
  # key="SampleID" means that column names (A1, A1, etc) are gathered into a new column
  # value="Quant" means the values in those columns are gathered into a new column
  # -c(UniprotAccession) leaves this column unchanged
  gather(key="SampleID", value="Quant", -c(UniprotAccession)) %>%
  # Create a copy of the SampleID column
  mutate(to_sep = SampleID) %>%
  # Select relevant columns
  select(UniprotAccession, SampleID, Quant)

# Log2 transformation
to_plot = to_plot %>%
  mutate(TransformQuant = log2(Quant))

knitr::kable(head(to_plot))
```

Check for right skewness.

```{r, message=FALSE}
# Plot distribution of raw data using density plot
ggplot(to_plot, aes(x=Quant)) +
  geom_density() +
  labs(title="Raw Data") +
  my_theme()

# Plot distribution of transformed data using density
ggplot(to_plot, aes(x=TransformQuant)) +
  geom_density() +
  labs(title="Transformed Data") +
  my_theme()
```

Before transformation, proteomics data have a highly skewed distribution with a long right tail.

After transformation, the data should look more symmetric and less skewed.

Check for large spread between minimum and maximum values, and possible outliers.

```{r, message=FALSE}
# Boxplot
ggplot(to_plot, aes(x=UniprotAccession, y=Quant)) +
  geom_boxplot() +
  labs(title="Raw Data") +
  my_theme() +
  theme(axis.text.x = element_blank())

# Boxplot
ggplot(to_plot, aes(x=UniprotAccession, y=TransformQuant)) +
  geom_boxplot() +
  labs(title="Transformed Data") +
  my_theme() +
  theme(axis.text.x = element_blank())
```

Before transformation, there may be a large spread between the minimum and maximum values, and possible outliers.

After log-transformation, the spread should be more compressed, and outliers may be less extreme.

Use a Q-Q plot to compare the distribution of your data to normal distribution.

```{r}
# Q-Q plot of raw data
qqnorm(to_plot$Quant)
qqline(to_plot$Quant, col="red")

# Q-Q plot of transformed data
qqnorm(to_plot$TransformQuant)
qqline(to_plot$TransformQuant, col="red")
```

Before transformation, the raw data will deviate significantly from the normal distribution, especially in the right tail.

After transformation, the data should show improve alignment with the normal distribution.

Check the mean-variance relationship. In proteomics data, variance typically increases with the mean intensity (i.e. the more abundant proteins show more variance).

```{r}
to_plot_mean_var = to_plot %>%
  group_by(UniprotAccession) %>%
  summarise(
    mean = mean(Quant, na.rm = TRUE),
    var = var(Quant, na.rm = TRUE)
  )

knitr::kable(head(to_plot_mean_var))
```

```{r}
# Plot mean vs variance for raw data
ggplot(to_plot_mean_var, aes(x=mean, y=var)) +
  geom_point(alpha=0.2) +
  labs(title="Raw Data") +
  my_theme()
```

```{r}
to_plot_mean_var = to_plot %>%
  group_by(UniprotAccession) %>%
  summarise(
    mean = mean(TransformQuant, na.rm = TRUE),
    var = var(TransformQuant, na.rm = TRUE)
  )

# Plot mean vs variance for transformed data
ggplot(to_plot_mean_var, aes(x=mean, y=var)) +
  geom_point(alpha=0.2) +
  labs(title="Transformed Data") +
  my_theme()
```

Log-transformation should stabilise the mean-variance relationship.

Overall, this data benefits from log-transformation as it improves normality and reduces skewness.

## Filtering missing data

Exclude proteins with high missingness - in this case, remove proteins with more than 30% missing values (NAs).

```{r}
# Filter for samples with NA_counts of 30 or less
to_retain = NAs %>%
  filter(NA_count < 31)

# Replace hypens with periods in the UniprotAccession column in TMT_data (to match to_retain)
TMT_data$UniprotAccession <- gsub("-", ".", TMT_data$UniprotAccession)
# Filter TMT_data to only include proteins found in to_retain
TMT_data_filtered = TMT_data %>%
  filter(UniprotAccession %in% to_retain$UniprotAccession)

nrow(TMT_data_filtered) # 6758
```

6758 proteins have quantification values in at least 70% of the samples.

## Batch normalisation

### Median normalisation

The protein intensities have been normalised via a multi-step process. Intensities were first normalised using the intensity measured for the bridge sample. An additional normalisation step performed by normalising the protein intensities measured for each sample by the global median of the per-sample median protein intensities.

Prepare the data for inspection of the normalisation by reshaping the dataset and extracting relevant information.

```{r}
# Select relevant data (non-pooled samples)
toplot = select(TMT_data_filtered, UniprotAccession, A1:J9) %>%
  # Reshape the data from wide format into long format (each row represents a single observation of a protein in sample)
  # key="SampleID" means that column names (A1, A1, etc) are gathered into a new column
  # value="Quant" means the values in those columns are gathered into a new column
  # -c(UniprotAccession) leaves this column unchanged
  gather(key="SampleID", value="Quant", -c(UniprotAccession)) %>%
  # Create a copy of the SampleID column
  mutate(to_sep = SampleID) %>%
  # Split the to_sep column into two new columns
  separate(to_sep, into=c("Batch", "Sample"), sep=1) %>%
  # Select relevant columns
  select(UniprotAccession, SampleID, Batch, Quant)

knitr::kable(head(toplot))
```

Plot the distribution of median-normalised protein quantification values per sample, grouped by batch.

```{r}
ggplot(data = toplot, aes(x=SampleID, y=Quant, color=Batch)) +
  geom_boxplot() +
  scale_y_log10() + # Scales y axis using log10 transformation to handle wide ranges in data
  my_theme() +
  theme(text=element_text(size=12)) + # Overall text size for plot
  theme(axis.text.x=element_text(angle=45, hjust=1, size=6)) + # x-axis text
  xlab("Sample") + # x-axis title
  ylab("Normalised protein quantification") # y-axis title
```

The boxplot allows inspection of potential batch effects. If the batches are well-balanced, the distributions should be similar.

Median normalisation has worked well across batches, but the overall distribution of protein quantification values between each sample is quite variable.

### Quantile normalisation

To resolve remaining variability in per-sample protein abundance, apply a full quantile normalisation.

```{r}
# Create a copy
TMT_data_filtered_quantile = TMT_data_filtered

library(preprocessCore)
# Normalise quantification values by standardising the distributions of quantification values across all samples
TMT_data_filtered_quantile[-c(1:5)] = normalize.quantiles(as.matrix(TMT_data_filtered_quantile[-c(1:5)]))

# Prepare data for plotting
toplot = select(TMT_data_filtered_quantile, UniprotAccession, UniprotProteinName, A1:J9) %>%
  gather(key = "SampleID", value = "Quant", -c(UniprotAccession, UniprotProteinName)) %>%
  mutate(to_sep = SampleID) %>%
  separate(to_sep, into = c("Batch", "Sample"), sep = 1) %>%
  select(UniprotAccession, SampleID, Batch, Quant)

# Plot
ggplot(data = toplot, aes(x=SampleID, y=Quant, color=Batch)) +
  geom_boxplot() +
  scale_y_log10() + # Scales y axis using log10 transformation to handle wide ranges in data
  my_theme() +
  theme(text=element_text(size=12)) + # Overall text size for plot
  theme(axis.text.x=element_text(angle=45, hjust=1, size=6)) + # x-axis text
  xlab("Sample") + # x-axis title
  ylab("Normalised protein quantification") # y-axis title
```

The distribution of quantification values across samples are more consistent.

## Filtering for synaptic proteins

As the Syn-PER process is an enrichment and not a purifications step, a second filter was applied to the proteins in the dataset using an inclusion list of 5667 synaptic proteins.

Load the data.

```{r}
fraction_lists = read_csv("../data/synaptic_inclusion_list.csv")

knitr::kable(head(fraction_lists))
```

Use the presynaptic and postsynaptic proteins in the list to filter the protein abundance data.

```{r}
# Extract postynaptic proteins
postsyn = unique(fraction_lists$postsyn_consensus_list)
# Extract presynaptic proteins
presyn = unique(fraction_lists$presyn_consensus)

# Filter the normalised data - keep rows where GeneName matches either a name in the postsyn or presyn list
TMT_data_synaptic = TMT_data_filtered_quantile %>%
  filter(GeneName %in% postsyn | GeneName %in% presyn)

knitr::kable(head(TMT_data_synaptic))
```

Export files.

```{r}
write_csv(TMT_data_synaptic, "../outs/TMT_data_synaptic.csv")
write_csv(TMT_data_filtered_quantile, "../outs/TMT_data_filtered_quantile.csv")
```
